internal/metrics/tx_mapper_db.go

```diff
@@ -439,78 +389,120 @@ func (tm *TxMapperDB) processTransactionExecution(
 	}
 
 	slot := te.DecKeysAndMessages[0].Slot
-	detailedBlock, err := tm.ethClient.BlockByNumber(ctx, big.NewInt(te.BlockNumber))
-	if err != nil {
-		return err
-	}
-
-	var blockTxHashes []common.Hash
-	for _, tx := range detailedBlock.Transactions() {
-		blockTxHashes = append(blockTxHashes, tx.Hash())
-	}
-
-	log.Debug().Int64("block number", detailedBlock.Number().Int64()).
-		Int64("slot", slot).
-		Msg("block info while processing decrypted transactions")
 
 	for index, txSubEvent := range txSubEvents {
-		decryptedTxHash, err := getDecryptedTXHash(txSubEvent, identityPreimageToDecKeyAndMsg)
-		if err != nil {
-			log.Err(err).Msg("error while trying to get decrypted tx hash")
-		}
 		decryptionKeyID, err := getDecryptionKeyID(txSubEvent, identityPreimageToDecKeyAndMsg)
 		if err != nil {
 			log.Err(err).Msg("error while trying to retrieve decryption key ID")
+			continue
 		}
-		if index < len(blockTxHashes) {
-			log.Debug().
-				Str("decryptedTXHash", decryptedTxHash.Hex()).
-				Str("blockTxHash", blockTxHashes[index].Hex()).
-				Bool("matches", decryptedTxHash.Cmp(blockTxHashes[index]) == 0).
-				Msg("comparing tx hash")
-			if decryptedTxHash.Cmp(blockTxHashes[index]) == 0 {
-				// it means we have it in correct order and the transaction is correct
-				err := tm.dbQuery.CreateDecryptedTX(ctx, data.CreateDecryptedTXParams{
-					Slot:                        slot,
-					TxIndex:                     txSubEvent.TxIndex,
-					TxHash:                      decryptedTxHash.Bytes(),
-					TxStatus:                    data.TxStatusValIncluded,
-					DecryptionKeyID:             decryptionKeyID,
-					TransactionSubmittedEventID: txSubEvent.ID,
-				})
-				if err != nil {
-					return err
-				}
-			} else {
-				// something went wrong case
-				err := tm.dbQuery.CreateDecryptedTX(ctx, data.CreateDecryptedTXParams{
-					Slot:                        slot,
-					TxIndex:                     txSubEvent.TxIndex,
-					TxHash:                      decryptedTxHash.Bytes(),
-					TxStatus:                    data.TxStatusValNotincluded,
-					DecryptionKeyID:             decryptionKeyID,
-					TransactionSubmittedEventID: txSubEvent.ID,
-				})
-				if err != nil {
-					return err
-				}
+		decryptedTx, err := getDecryptedTXHash(txSubEvent, identityPreimageToDecKeyAndMsg)
```

Comment:
according to this function it returns the tx hash. how can you send a transaction with purely passing the hash?

Reply:
it previously used to return tx hash but now it returns  `*types.Transaction`. Need to change the name of the function to `getDecryptedTX`